src/openssl.cr

```diff
@@ -1,69 +1,99 @@
 require "./openssl/lib_ssl"
 require "./openssl/error"
 
-# NOTE: To use `OpenSSL`, you must explicitly import it with `require "openssl"`
+# The OpenSSL module allows for access to Secure Sockets Layer (SSL) and Transport Layer Security (TLS)
+# encryption, as well as classes for encrypting data, decrypting data, and computing hashes. It uses
+# the SSL library provided by the operating system, which may be either [OpenSSL](https://openssl.org)
+# or [LibreSSL](https://www.libressl.org).
 #
-# ## OpenSSL Integration
+# WARNING: This module should not be used without first reading the [Security considerations](#security-considerations).
 #
-# - TLS sockets need a context, potentially with keys (required for servers) and configuration.
-# - TLS sockets will wrap the underlying TCP socket, and any further communication must happen through the `OpenSSL::SSL::Socket` only.
+# To can create secure sockets, use either `OpenSSL::SSL::Socket::Client` for client applications, or
+# `OpenSSL::SSL::Socket::Server` for servers.  These classes use a default context, but you can provide
+# your own by supplying an `OpenSSL::SSL::Context`.  For more control, consider subclassing `OpenSSL::SSL::Socket`.
 #
-# ## Usage Example
+# Hashing algorithms are provided by classes such as `Digest::SHA256` and `Digest::MD5`.  If you need
+# a different option, you can initialize one using the name of the digest with `OpenSSL::Digest`.
+# A Hash-based Message Authentication Code (HMAC) can be computed using `HMAC` and specifying a digest
+# `Algorithm`.
+#
+# The `OpenSSL::Cipher` class can be used for encrypting and decrypting data.
+#
+# NOTE: To use `OpenSSL`, you must explicitly import it using the `require "openssl"` statement.
+#
+# ## Security Considerations
+#
+# Crystal aims to provide reasonable configuration defaults in accordance with
```

Comment:
We may want to _update_ the reference to OWASP rather than remove it:

https://cheatsheetseries.owasp.org/cheatsheets/Transport_Layer_Security_Cheat_Sheet.html#server-configuration

Reply:
Oh nevermind, I missed it :facepalm: